### PR TITLE
(332) Add docs for managing infrastructure

### PR DESCRIPTION
**Pull request by Gweaton**
## Changes in this PR

Adds docs for managing infrastructure with reference to tools like Kubernetes, Kibana and the MOJ Cloud Platform Environments project.

This includes instructions for viewing logs, debugging deployments, restarting services etc.

It's very much copied and pasted/adapted from the Community Accommodation team's guidance
[here](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4325244964/Manage+infrastructure)

## Post-merge checklist

- [ ] Have you written an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e) (if necessary)?

### CODE REVIEW

File: doc/how-to/manage-infrastructure.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+# Manage Infrastructure
```
Comment (yndajas) [inline doc/how-to/manage-infrastructure.md:1]: This is title case whereas the other file's `h1` and all other headings are sentence case

File: doc/how-to/manage-infrastructure.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+When this app deploys via merging to `main`, CircleCI will automatically
+propagate those changes to the cluster. This happens through the
+`hmpps/deploy_env` job that is provided by the [MoJ CircleCI
+Orb](https://github.com/ministryofjustice/hmpps-circleci-orb).
+
+Each environment will correspond to a Cloud Platform 'namespace'. The namespace
+is an isolated cluster. We can use the [Cloud Platform Environments
+repository](https://github.com/ministryofjustice/cloud-platform-environments/tree/main/namespaces/live.cloud-platform.service.justice.gov.uk)
+to define our backing services, certificates etc.
```
Comment (yndajas) [inline doc/how-to/manage-infrastructure.md:19]: Are these paragraphs the wrong way around? `The namespace is an isolated cluster` (which sounds like a definition of a cluster?) comes after `propagate those changes to the cluster` (which is referring to "the cluster" before it's defined)

File: doc/how-to/manage-infrastructure.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+## Prerequisites
```
Comment (yndajas) [inline doc/how-to/manage-infrastructure.md:21]: Prerequisites for what? Connecting to Kubernetes? Maybe this section should be `Connecting to Kubernetes`, with the first two points as prerequisites and the last as a link to instructions? (Unless you want to have a prerequisites section on both docs)

File: doc/how-to/manage-infrastructure.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+## Kubernetes cheat sheet
+
+To use Kubernetes to interact with the cluster there's [a cheat
+sheet](https://kubernetes.io/docs/reference/kubectl/cheatsheet/). Keep reading
+for tasks we commonly use.
+
+## Common Kubernetes tasks
+
+### View the application logs of a pod
```
Comment (yndajas) [inline doc/how-to/manage-infrastructure.md:38]: Is a pod different to a cluster?

File: doc/how-to/view-application-logs.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+- Be a part of the Ministry of Justice GitHub organisation. This should be done
+  as part of your your team onboarding process.
+
+## Kibana
+
+While you can [view the logs of an individual Kubernetes
+pod](./manage-infrastructure.md#view-the-application-logs-of-a-pod), the logs
+can also be viewed in the interface on
+[Kibana](https://kibana.cloud-platform.service.justice.gov.uk/)
```
Comment (yndajas) [inline doc/how-to/view-application-logs.md:13]: Full stop

File: doc/how-to/manage-infrastructure.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+to define our backing services, certificates etc.
+
+## Prerequisites
+
+- Be a part of the Ministry of Justice GitHub organisation. This should be done
+  as part of your your team onboarding process.
+- Be a part of the `accredited-programmes-team` team. This should be done as
+  part of your team onboarding process.
+- [Follow the Cloud Platform guidance to connect to the Kubernetes
+  cluster](https://user-guide.cloud-platform.service.justice.gov.uk/documentation/getting-started/kubectl-config.html#connecting-to-the-cloud-platform-39-s-kubernetes-cluster)
```
Comment (yndajas) [inline doc/how-to/manage-infrastructure.md:28]: Full stop

**Review by yndajas — APPROVED** [review]
Looks good. Just a few proofreading comments

File: doc/how-to/manage-infrastructure.md (hunk 4 above)
Comment (Gweaton) [reply doc/how-to/manage-infrastructure.md:38]: Yes - [this documentation](https://kubernetes.io/docs/concepts/overview/components/) might be a helpful overview.

TL;DR a "cluster" is the thing that's deployed, which is a collection of "nodes" (worker machines) which host "pods" that run dockerised applications e.g. our application & the API in this case. So when we're debugging, we want to look at the output of these pods as they're what our app is running in.

File: doc/how-to/manage-infrastructure.md (hunk 2 above)
Comment (Gweaton) [reply doc/how-to/manage-infrastructure.md]: Yeah that's a fair point. I've swapped them round and I think it makes more sense this way.

File: doc/how-to/manage-infrastructure.md (hunk 6 above)
Comment (Gweaton) [reply doc/how-to/manage-infrastructure.md:28]: Oops, I thought I'd fixed this but realised it hadn't changed. Sorry, this'll have to be without a full stop for a bit...

Comment (yndajas) [reply doc/how-to/manage-infrastructure.md:28]: I think we'll cope
